crates/credentials/src/vc.rs

```diff
@@ -90,13 +126,69 @@ impl VerifiableCredential {
 
         Ok(jwt)
     }
-
     pub async fn verify(jwt: &str) -> Result<Self, CredentialError> {
         let jwt_decoded = Jwt::verify::<VcJwtClaims>(jwt).await?;
 
-        // TODO Implement semantic VC verification rules https://github.com/TBD54566975/web5-rs/issues/151
+        let mut vc = jwt_decoded.claims.vc;
 
-        Ok(jwt_decoded.claims.vc)
+        validate_vc_data_model(&vc).map_err(CredentialError::ValidationError)?;
+
+        // exp MUST represent the expirationDate property, encoded as a UNIX timestamp (NumericDate).
+        if let Some(exp) = jwt_decoded.claims.registered_claims.expiration {
+            if let Some(expiration_date) = vc.expiration_date {
+                if exp != expiration_date {
+                    return Err(CredentialError::ExpirationDateMismatch);
+                }
+            } else {
+                vc.expiration_date = Some(exp);
```

Comment:
Help me understand why we would want to set values on the `vc` here; I'm imagining we're doing this in other SDKs so I'm ignorant to the reasoning.

The reason I raise it is, it's a slightly weird developer experience to me, for a caller to pass in a JWT-encoded representation of a VC which has a set of properties A and then be handed the decoded VC with a set of properties B (that is, a different set of properties than what's in the JWT-encoded representation)

Reply:
![image](https://github.com/TBD54566975/web5-rs/assets/5314059/c6e16ca6-0937-45a8-869b-f0ead05abfad)

the "official official" spec says that if exp is there you don't have to have vc.exirationDate.

To be explicit we want both, but to be compliant to spec we have a case if exp is there but there is no vc.exirationDate then we set it to exp 

but if they are out of sync in any other way then it fails

Reply:
its only like this for optional fields so its a slightly different truth table for required fields (basically it should have both and they should match)

I'll go through and double check I'm in sync with these truth tables

Reply:
^ check new pr message explaining this a bit better

Reply:
oh interesting this is at the spec level! okay great well done 👏